applications/GeoMechanicsApplication/custom_elements/small_strain_U_Pw_diff_order_element.cpp

```diff
@@ -1042,9 +1042,14 @@ void SmallStrainUPwDiffOrderElement::
 
     // KRATOS_INFO("0-SmallStrainUPwDiffOrderElement::1-SetValuesOnIntegrationPoints()") << std::endl;
 
-    for ( unsigned int GPoint = 0; GPoint < mConstitutiveLawVector.size(); ++GPoint )
-        mConstitutiveLawVector[GPoint]->SetValue( rVariable, rValues[GPoint], rCurrentProcessInfo );
-
+        for (unsigned int GPoint = 0; GPoint < mConstitutiveLawVector.size(); ++GPoint) {
+            if (rVariable == CAUCHY_STRESS_VECTOR) {
+               mStressVector[GPoint] = rValues[GPoint];
+            }
+            else {
+                mConstitutiveLawVector[GPoint]->SetValue(rVariable, rValues[GPoint], rCurrentProcessInfo);
+            }
+        }
```

Comment:
If I'm not mistaken, the point here was that we needed different behavior for the Cauchy stress vector. The current implementation tests this for every iteration in the `for` loop, whereas `rVariable` won't change inside of the loop. Perhaps we can express this more directly as follows:
```suggestion
        if (rVariable == CAUCHY_STRESS_VECTOR) {
            std::copy(rValues.begin(), rValues.end(), mStressVector.begin());
        }
        else {
            for (unsigned int GPoint = 0; GPoint < mConstitutiveLawVector.size(); ++GPoint) {
                mConstitutiveLawVector[GPoint]->SetValue(rVariable, rValues[GPoint], rCurrentProcessInfo);
            }
        }
```
What do you think?

Reply:
Incorporated, with length checks for vector sizes.